yt_dlp/extractor/on24.py

```diff
@@ -11,12 +11,15 @@ class On24IE(InfoExtractor):
     IE_NAME = 'on24'
     IE_DESC = 'ON24'
 
-    _VALID_URL = r'''(?x)
-                    https?://event\.on24\.com/(?:
-                        wcc/r/(?P<id_1>\d{7})/(?P<key_1>[0-9A-F]{32})|
-                        eventRegistration/(?:console/EventConsoleApollo|EventLobbyServlet\?target=lobby30)
-                            \.jsp\?(?:[^/#?]*&)?eventid=(?P<id_2>\d{7})[^/#?]*&key=(?P<key_2>[0-9A-F]{32})
-                    )'''
+    _ID_RE = r'(?P<id>\d{7})'
+    _KEY_RE = r'(?P<key>[0-9A-F]{32})'
+    _URL_BASE_RE = r'https?://event\.on24\.com'
+    _URL_QUERY_RE = rf'(?:[^#]*&)?eventid={_ID_RE}&(?:[^#]+&)?key={_KEY_RE}'
+    _VALID_URL = [
+        rf'{_URL_BASE_RE}/wcc/r/{_ID_RE}/{_KEY_RE}',
+        rf'{_URL_BASE_RE}/eventRegistration/console/(?:EventConsoleApollo\.jsp|apollox/mainEvent/?)\?{_URL_QUERY_RE}',
+        rf'{_URL_BASE_RE}/eventRegistration/EventLobbyServlet/?\?target=[\w-]+\.jsp&{_URL_QUERY_RE}',
```

Comment:
why bother with the target param even
```suggestion
        rf'{_URL_BASE_RE}/eventRegistration/EventLobbyServlet/?\?{_URL_QUERY_RE}',
```